src/lib/uri/uri.h

```diff
@@ -23,6 +32,10 @@ struct uri {
         char *query;
         char *fragment;
         int host_hint;
+        /** Count of URI options */
+        int option_count;
+        /** Different URI options */
+        struct uri_query_param *options;
```

Comment:
They are called 'parameters' now, not options. Please update these variable names to `param_count` and `params` (or `query_param_count` and `query_params`) and also update other variables and comments throughout the code.

Come to think of it, `uri_query_param` is a somewhat long name. Maybe, use `uri_param`, `uri_param_count`, `uri_param_add_value`? Or it would be confusing? Up to you - I'm fine with either way, provided names are consistent throughout the code. I assume that `struct uri_query_param param` is consistent, as well as `struct uri_query_param query_param`, but `struct uri_query_param option` is not.